pyproject.toml

```diff
@@ -23,7 +23,7 @@ classifiers = [
   "Programming Language :: Python :: 3",
   "Topic :: Software Development :: Libraries :: Python Modules",
 ]
-dependencies = ["sphinx", "packaging", "requests", "lxml"]
+dependencies = ["sphinx<9", "packaging<25", "requests<3", "lxml<6"]
```

Comment:
Why are you adding an upper-bound pin here?

Reply:
I want to add pins because, when installing without pixi (like on our conda-forge feedstock), and in the case a dependency publishes a major release with probable breaking changes (like sphinx did), we can't safely depend on these.

I added upper bound pins because we still mostly support older versions (e.g. sphinx 7) and setting a _compatible pin_ (`sphinx~=8.0`) would force the use of the new packages (sphinx 8).
I could change them to _lesser or equal to_ specifiers instead (`sphinx<=8`).
Or if you think it is too strict, I'll remove these, no worries.

Reply:
I think this strategy does not pay at the long term for a "rolling release" version management such as in conda-forge.  It will impede a bunch of software stacks to be updated for a bunch of users.  So, unless there is a **very** compelling reason on why you're down-pinning 4 packages, you should refrain from doing that.

In general, it's best to "risk" an upgrade instead of jeopardising it.

With that philosophy in mind, can you please adjust the pins so they are as loose as possible, however constrained as minimally required so that the package still runs w/o errors?

Reply:
For example, you mention a problem with sphinx.  Fine - add a pin on it.  Is it only on version 8 (can you say something like `!=8`?  Or is it above 8?).  However, please remove all others if they are not absolutely required.